src/pages/packer/plugins/index.tsx

```diff
@@ -36,12 +37,17 @@ async function getStaticProps(ctx) {
 		// Partial nav data is provided from base getStaticProps, in menuItems
 		const partialNavData =
 			staticProps.props.layoutProps.sidebarNavDataLevels[2].menuItems
-		// Fetch and merge in remote plugins nav data with the partialNavData
-		const rawNavData = await appendRemotePluginsNavData(
-			remotePluginsFile,
-			partialNavData,
-			''
-		)
+
+		let rawNavData = partialNavData
+		if (!isDeployPreview() || isDeployPreview('packer')) {
```

Comment:
I'm not happy with some of these new conditionals. An alternative would be to remove the actual page files for unrelated products, like we do for the io previews.

Reply:
if i'm reading right, is the unhappiness here also related to `/packer/plugins` (& `/boundary/api-docs` to an extent) not having their `server` stuff as well abstracted as other docs routes?

Reply:
A bit! Maybe it won't be an issue in practice, but anytime we're relying on data from another repo through the GH API, we'll likely need a conditional like this. It also just feels awkward coming upon these conditionals without the right context 💭  There are three states that we realistically should handle for every case:

* Running in dev-portal, always remote
* Running in a content deploy preview, but not the source of the data (remote)
* Running in a content deploy preview, in the source of the data (local)

I think we're safe for now, but it _feels_ like there could be an abstraction to be created here...

Reply:
☝️ This feels like really helpful context. Should we document this in a README that explains how docs content is handled in dev dot or add a comment here?

Reply:
Great idea!

Reply:
@kendallstrautman I dropped a section in the repo docs: https://github.com/hashicorp/dev-portal/blob/brk.feat/developer-preview/docs/README.md#remote-content--application-context